src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -48,6 +48,76 @@ static class BitOperations
             19, 27, 23, 06, 26, 05, 04, 31
         };
 
+        /// <summary>
+        /// Evaluate whether a given integral value is a power of 2.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsPow2(int value)
+        {
+            if (Popcnt.IsSupported)
+            {
+                // 1 set bit means number is a power of 2 unless it is the sign bit, so get rid of that
+                // Cast to unsigned type so logic shift not signed arithmetic shift
+                return PopCount((uint)value << 1) == 1;
+            }
+
+            return IsPow2SoftwareFallback((uint)value << 1);
```

Comment:
This should be a static local function rather than a private function external to `IsPow2`